src/mscorlib/src/System/CurrentSystemTimeZone.cs

```diff
@@ -29,12 +29,8 @@ namespace System
 {
     [Obsolete("System.CurrentSystemTimeZone has been deprecated.  Please investigate the use of System.TimeZoneInfo.Local instead.")]
     [Serializable]
-    internal class CurrentSystemTimeZone : TimeZone
+    internal partial class CurrentSystemTimeZone : TimeZone
     {
-        // The per-year information is cached in in this instance value. As a result it can
-        // be cleaned up by CultureInfo.ClearCachedData, which will clear the instance of this object
```

Comment:
Wait, it looks like it is clearing it -  CultureInfo.ClearCachedData() calls TimeZone.ResetTimeZone which does what it always has - clear out the cached "currentTimeZone." which eventually orphans the table along with the current timezone.